app/src/org/commcare/tasks/LogSubmissionTask.java

```diff
@@ -268,7 +272,9 @@ private static LogSubmitOutcomes submitDeviceReportRecord(DeviceReportRecord slr
             return LogSubmitOutcomes.SUBMITTED;
         }
 
-        listener.startSubmission(index, f.length());
+        if (listener != null) {

```

Comment:
nit: add `@Nullable` to the global variable declation for `listener` to signify it can be `null`